pypesto/petab/importer.py

```diff
@@ -443,6 +457,17 @@ def create_problem(
         if objective is None:
             objective = self.create_objective(**kwargs)
 
+        x_fixed_indices = self.petab_problem.x_fixed_indices
+        x_fixed_vals = self.petab_problem.x_nominal_fixed_scaled
+        x_ids = self.petab_problem.x_ids
+        if isinstance(objective.calculator, HierarchicalAmiciCalculator):
+            for inner_x_id in objective.calculator.inner_problem.get_x_ids():
+                if inner_x_id in x_fixed_indices:
+                    x_fixed_vals[x_fixed_indices.index(inner_x_id)] = np.nan
```

Comment:
these edits are NOT inplace, so they have no effect. It looks like optimization is still performed over all parameters, including those specified as hierarchical.